[quant][api] Expose graph mode quantization API in `torch.quantization`

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40212 [quant][graphmode] Rename graph mode quantization API to `quantize_jit`
* **#40198 [quant][api] Expose graph mode quantization API in `torch.quantization`**
* #40197 [quant][graphmode] Add support for detach
* #40196 [quant][graphmode] Clean up and add more logging

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22106542](https://our.internmc.facebook.com/intern/diff/D22106542)

jerryzh168 committed:
[quant][api] Expose graph mode quantization API in `torch.quantization`

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/quantization/__init__.py:
```diff
@@ -18,9 +18,11 @@ def default_eval_fn(model, calib_data):
-    'prepare', 'convert',
+    'quantize', 'quantize_dynamic', 'quantize_qat',
+    'prepare', 'convert', 'prepare_qat',
+    # Top level API for graph mode quantization
```
just curious, if the name is "graph mode", why not call the APIs with `_graph` instead of `_script`?

jerryzh168 replied:
`_script` is referring to `TorchScript`, indicating it's taking a TorchScript model. 
I think eager mode and graph mode is a separate(broader) naming convention, that is, other frameworks will have eager mode/graph mode as well, we don't really have these appear in the APIs.

supriyar replied:
@jerryzh168 do you think changing the name to `quantize_torchscript` would be a good idea? `_script` could have a different connotation that we are scripting the model and quantizing it.

vkuzo replied:
for me "script" triggers "this is a script" instead of "this is TorchScript" as a first response.  Don't feel super strongly, but naming it more descriptively might be valuable.

jerryzh168 replied:
that is too long I think. I think people still need to read the doc to know how to use it anyways, the name shouldn't be that important.

jerryzh168 replied:
what about `quantize_jit` or `quantize_ts`?

vkuzo replied:
> quantize_jit

I really like that one, because JIT in the context of PyTorch is very specific

jerryzh168 replied:
sounds good, will put up a rename PR

jerryzh168 committed:
Update on "[quant][api] Expose graph mode quantization API in `torch.quantization`"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][api] Expose graph mode quantization API in `torch.quantization`"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][api] Expose graph mode quantization API in `torch.quantization`"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22106542](https://our.internmc.facebook.com/intern/diff/D22106542)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][api] Expose graph mode quantization API in `torch.quantization`"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22106542](https://our.internmc.facebook.com/intern/diff/D22106542)

[ghstack-poisoned]
